submissions/starter-code/scripts/articleView.js

```diff
@@ -0,0 +1,148 @@
+// Configure a view object, to hold all our functions for dynamic updates and article-related event handlers.
+var articleView = {};
+
+articleView.populateFilters = function() {
+  $('article').each(function() {
+    if (!$(this).hasClass('template')) {
+      // DONE: We need to take every author name from the page, and make it an option in the Author filter.
+      //       To do so, we build an `option` DOM element that we can append to the author select box.
+      //       Start by grabbing the author's name from `this` article element, and then use that bit of
+      //       text to create the option tag (in a variable named `optionTag`),
+      //       that we can append to the #author-filter select element.
+      //       YAY, DOM manipulation!
+      var val = $(this).find('address a').text();
+      var optionTag = '<option value="' + val + '">' + val + '</option>';
+      $('#author-filter').append(optionTag);
+
+      // DONE: Similar to the above, but...
+      //       Avoid duplicates! We don't want to append the category name if the select
+      //       already has this category as an option!
+      val = $(this).attr('data-category');
+      optionTag = '<option value="' + val + '">' + val + '</option>';
+      if ($('#category-filter option[value="' + val + '"]').length === 0) {
+        $('#category-filter').append(optionTag);
+      }
+    }
+  });
+};
+
+articleView.handleAuthorFilter = function() {
+  $('#author-filter').on('change', function() {
+    if ($(this).val()) {
+      // TODO: If the select box was changed to an option that has a value, we need to:
+      //       1. Hide all the articles,
+      //       2. Fade in only the articles that match based on the author that was selected.
+      //          Use an "attribute selector" to find those articles that match the value,
+      //          and fade them in for the reader.
+      $('article').hide();
+      var filterValue = $(this).val();
+      $('article').each(function() {
```

Comment:
This works, but a loop is an inefficient way to handle it. Look at jQuery selectors, is there one you can use to select the right element immediately?  https://api.jquery.com/category/selectors/